### PR TITLE
Repos without service

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/clients/repositories.rb [hunk 1]
```diff
@@ -171,6 +171,7 @@ def PriorityToString(priority)
 
     def ReposFromService(service, input)
       input = deep_copy(input)
+      service = "" if service == :without
```
Comment (lslezak) [inline src/clients/repositories.rb]: `:without` is used at several places, we should define a constant for it.

(And `:without` symbol sounds a bit strange for me, I'd rather prefer `:no_service` or something like that.)

File: src/clients/repositories.rb [hunk 2]
```diff
@@ -641,10 +642,20 @@ def buildList
...
         )
         ret = Builtins.add(ret, t)
-      end 
+      end
 
+      # there is some service, so allow to filter repos without service (bnc#944504)
+      if ret.size > 2
+        t = Item(
+          Id(:without_service),
+          _("Without Service"),
```
Comment (lslezak) [inline src/clients/repositories.rb]: It's the other way round, a service contains repositories. The filter should be named "Repositories not in a service",  "Repositories not belonging to a service" or something like this. A bit longer but I'm not sure if there is a good shorter alternative...

@mvidner any suggestion here?

File: package/yast2-packager.changes [hunk 3]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Tue Oct  6 11:13:46 UTC 2015 - jreidinger@suse.com
+
+- If there is service allow to see repositories without service
```
Comment (mvidner) [inline package/yast2-packager.changes]: > If there are services, allow to see repositories not belonging to a service

Comment (mvidner) [inline package/yast2-packager.changes]: Ah, only now I understood the bug report.
> Added a filter showing only those repositories not belonging to a service.

File: src/clients/repositories.rb (hunk 2 above)
Comment (mvidner) [inline src/clients/repositories.rb]: "Only repositories not provided by a service"

Comment (kobliha) [inline src/clients/repositories.rb]: And then definitely add # TRANSLATORS: Explanation what it actually means :) and where it is used

File: src/clients/repositories.rb [hunk 4]
```diff
@@ -641,10 +642,20 @@ def buildList
             @displayed_service == Ops.get_string(srv_state, "alias", "")
         )
         ret = Builtins.add(ret, t)
-      end 
+      end
 
+      # there is some service, so allow to filter repos without service (bnc#944504)
+      if ret.size > 2
```
Comment (mvidner) [inline src/clients/repositories.rb]: `size > 2` is very non-obvious. how about `unless Builtins.isempty(@serviceStatesOut)`

File: src/clients/repositories.rb (hunk 2 above)
Comment (jreidinger) [inline src/clients/repositories.rb]: OK, will add TRANSLATORS, just improved a bit mvidner text with replaced "a" with "any" as there is usually more services there, so not provided by any service.

File: src/clients/repositories.rb (hunk 1 above)
Comment (jreidinger) [inline src/clients/repositories.rb]: will fix

File: src/clients/repositories.rb (hunk 4 above)
Comment (jreidinger) [inline src/clients/repositories.rb]: that is reason why I add comment there. Basically I am not fan of adding new buildins and without it we need to be very care to not cause exception

**Comment by lslezak** [comment]
LGTM, do not forget to send the new POT file to Karl, see the "Text Freeze, SP1" mail on yast-devel@.

**Pull request merged**